src/controller/controller.ts

```diff
@@ -683,7 +683,13 @@ class Controller extends events.EventEmitter {
             if (frame.isGlobal()) {
                 if (frame.isCommand('report')) {
                     type = 'attributeReport';
-                    data = ZclFrameConverter.attributeKeyValue(dataPayload.frame);
+                    // Certain devices (e.g. Legrand/4129) fail to set the manufacturerSpecific flag and
```

Comment:
Please move these comments to `zclFrameConverters.ts`, the only change needed here is:

```js
data = ZclFrameConverter.attributeKeyValue(dataPayload.frame, device.manufacturerID);
```